tools/bootstrap_commitlogs/main/main.go

```diff
@@ -0,0 +1,233 @@
+	return strconv.Itoa(int(unixTs))
+}
+
+func main() {
+	flag.Parse()
+	if *pathPrefixArg == "" ||
+		*namespaceArg == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	var (
+		pathPrefix           = *pathPrefixArg
+		namespaceStr         = *namespaceArg
+		blockSize            = *blockSizeArg
+		flushSize            = *flushSizeArg
+		bootstrapRetention   = *bootstrapRetentionArg
+		shardsCount          = *shardsCountArg
+		shards               = *shardsArg
+		debugListenAddress   = *debugListenAddressArg
+		currentUnixTimestamp = *currentUnixTimestampArg
+	)
+
+	log := xlog.NewLogger(os.Stderr)
+
+	if debugListenAddress != "" {
+		go func() {
+			log.Infof("starting debug listen server at '%s'\n", debugListenAddress)
+			err := http.ListenAndServe(debugListenAddress, http.DefaultServeMux)
+			if err != nil {
+				log.Fatalf("could not start debug listen server at '%s': %v", debugListenAddress, err)
+			}
+		}()
+	}
+
+	shardTimeRanges := result.ShardTimeRanges{}
+
+	currentUnixTimestampParsed, err := strconv.Atoi(currentUnixTimestamp)
+	if err != nil {
+		log.Fatalf("could not parse unix timestamp: '%s': %v", currentUnixTimestamp, err)
+	}
+
+	now := time.Unix(int64(currentUnixTimestampParsed), 0)
+	// Round current time down to nearest blocksize (2h) and then decrease blocksize (2h)
+	startInclusive := now.Truncate(blockSize).Add(-blockSize)
+	// Round current time down to nearest blocksize (2h) and then add blocksize (2h)
+	endExclusive := now.Truncate(blockSize).Add(blockSize * 2)
+
+	// Ony used for logging
+	var shardsAll []uint32
+
+	// Handle commda-delimited shard list 1,3,5, etc
+	if strings.TrimSpace(shards) != "" {
+		for _, shard := range strings.Split(shards, ",") {
+			shard = strings.TrimSpace(shard)
+			if shard == "" {
+				log.Fatalf("Invalid shard list: '%s'", shards)
+			}
+			value, err := strconv.Atoi(shard)
+			if err != nil {
+				log.Fatalf("could not parse shard '%s': %v", shard, err)
+			}
+			rng := xtime.Range{Start: startInclusive, End: endExclusive}
+			shardTimeRanges[uint32(value)] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, uint32(value))
+		}
+		// Or just handled up to N (shard-count) shards
+	} else if shardsCount > 0 {
+		for i := uint32(0); i < uint32(shardsCount); i++ {
+			rng := xtime.Range{Start: startInclusive, End: endExclusive}
+			shardTimeRanges[i] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, i)
+		}
+	} else {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	log.WithFields(
+		xlog.NewField("pathPrefix", pathPrefix),
+		xlog.NewField("namespace", namespaceStr),
+		xlog.NewField("shards", shardsAll),
+	).Infof("configured")
+
+	instrumentOpts := instrument.NewOptions().
+		SetLogger(log)
+
+	retentionOpts := retention.NewOptions().
+		SetBlockSize(blockSize).
+		SetRetentionPeriod(bootstrapRetention).
+		SetBufferPast(1 * time.Minute).
+		SetBufferFuture(1 * time.Minute)
+
+	blockOpts := block.NewOptions()
+
+	encoderPool := encoding.NewEncoderPool(nil)
```

Comment:
might be worthwhile to set the defaults to the pool options to the same thing we have for the nodes in production

Reply:
Is that in a yaml file or...? How do i find that? Or is it just the default options and I just need to instantiate it?

Reply:
Found em

Reply:
done